src/greedy_match.cpp

```diff
@@ -31,8 +28,8 @@ List greedy_string(SEXP radix, CharacterVector to_match){
       if(holding.size() == 0){
         holding.push_back(NA_STRING);
```

Comment:
Thought; could we simplify this? instead of checking holding.size(), add `if(it = vec.end()` as a check before looping through the iterator.

Again, not required in this PR, just taking the opportunity to run shit by you.